File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    hash := map[int]int{}
    
    for i, val := range nums {
        num := target-val
        
        if x, in := hash[num]; in {
            return []int{x, i}
        } else {
            hash[val] = i
        }
    }
    
    return []int{}
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if not nums:
            return []

        table = {}
        for i, v in enumerate(nums):
            n = target - v
            if n not in table:
                table[v] = i 
            else:
                return [i, table[n]]